lasagne/layers/shape.py

```diff
@@ -309,11 +314,20 @@ def __init__(self, incoming, width, val=0, batch_ndim=2, **kwargs):
 
     def get_output_shape_for(self, input_shape):
         output_shape = ()
+        l = [0] * len(input_shape)
+        r = [0] * len(input_shape)
         for k, s in enumerate(input_shape):
             if k < self.batch_ndim:
                 output_shape += (s,)
             else:
-                output_shape += (s + 2 * self.width,)
+                try:
+                    l[k], r[k] = self.width[k - self.batch_ndim]
+                except TypeError:
+                    try:
+                        l[k] = r[k] = self.width[k - self.batch_ndim]
+                    except TypeError:
+                        l[k] = r[k] = self.width
+                output_shape += (input_shape[k] + l[k] + r[k],)
```

Comment:
Using lists `l` and `k` when you're just interested in the current value seems a bit wasteful. What about:
```python
def get_output_shape_for(self, input_shape):
    output_shape = list(input_shape)
    if isinstance(self.width, int):
        widths = [self.width] * (len(input_shape) - self.batch_ndim)
    else:
        widths = self.width
    for k, w in enumerate(self.width):
        try:
            l, r = w
        except (TypeError, ValueError):
            l = r = w
        output_shape += l + r
```
This also handles arbitrary iterables for `self.width`. Note that your solution did not catch the `ValueError` for a wrong-length tuple.